### ISSUE TITLE
Slash Commands Not Available

### ISSUE THREAD

**Issue by brodkin** [demodude4u/Factorio-FBSR#142]
Installed the bot, but all of the commands say to use slash commands instead. Unfortunately, even after uninstalling and reinstalling, the slash commands do not populate into Discord.

![Screen Shot 2022-05-02 at 10 04 39](https://user-images.githubusercontent.com/236564/166247490-e97340dc-ab84-442f-8cd4-3394eb298336.png)

![Screen Shot 2022-05-02 at 10 02 50](https://user-images.githubusercontent.com/236564/166247491-af1b52ad-291b-4c1e-bbc7-56f57639791e.png)

Tested in web interface and Discord Canary v0.0.284

**Comment by demodude4u** [comment]
I have seen some servers configured to disable slash commands.  Double-check that this isn't the case with this issue.  Sometimes people will disable slash commands because they do not want admin commands from another bot to appear.  If this does not resolve the issue, can you verify that you can see the commands in a brand new server?

**Comment by brodkin** [comment]
Slash commands are definitely enabled. You can actually see ProBot’s in the screenshot above. I tried spinning up a new server and adding the bot there but same result unfortunately.

**Comment by Bilka2** [comment]
Your screenshot is missing the "command permissions" section that is shown here: https://discord.com/blog/slash-commands-permissions-discord-apps-bots. Maybe it's somewhere else due to discord canary? Try to find that section and see if the commands are allowed there.

**Comment by brodkin** [comment]
Yeah, I was including that screenshot to demonstrate that the commands are missing. ProBot has them listed, even in Canary.

 I am also a Discord dev so I started looking over the docs and it looks like the invite link needs to include the "applications.commands" scope in the dev portal's OAuth2 URL Generator for command permission to be granted. Perhaps the two invite links I tried are missing that?

**Comment by Bilka2** [comment]
Maybe. Try https://discord.com/oauth2/authorize?client_id=310607046020169729&scope=bot+applications.commands&permissions=379968, it's the invite from the /info command. It has the applications.commands scope and works on my server.

**Comment by brodkin** [comment]
That's it! I previously used the invite links from top.gg and Factorio Cheat Sheet, but neither one worked. This one was added successfully and works as expected.

I also tried using the below URL to just authorize commands, which added successfully, but errored on use. In theory, it's all that would be needed for a bot that only responds to commands though. Perhaps an enhancement?
https://discord.com/api/oauth2/authorize?client_id=310607046020169729&scope=applications.commands 

In any event, I appreciate your help! It's unfortunate that the scopes/permissions are in the URL itself since you don't have control of everywhere the invite link may exist.

Perhaps that is why this new system is starting in August...
https://support-dev.discord.com/hc/en-us/articles/4404772028055
